### PR TITLE
BIGTOP-3456: Add hadoop-yarn-router and hadoop-hdfs-dfsrouter to hadoop services

### PR DESCRIPTION
**Pull request by qizhu-lucas**
I have used bigtop to release our production cluster based hadoop 3.2.1(with some 3.3.0 patchs fixed), and some hadoop env related not fixed in default.
1.  Fix  start/stop error service pid
2. Fix historyserver/timelineserver/proxyserver PIDFILE (affect start failed).
3. Add yarn federation router server support.
4. Support YARN new web ui.
5. Support for hdfs router.

### CODE REVIEW

**Comment by guyuqi** [comment]
Thank you for working on it. 
As mentioned in  BIGTOP-3280, we plan to upgrade Hadoop to 3.3.0.

But anyway, nice to meet you in Bigtop. 
It would be appreciated if you join us and contribute more. 
Pls join our mailing list: https://bigtop.apache.org/mail-lists.html, thanks.

**Comment by iwasakims** [comment]
@qizhu-lucas Sorry for late reply. This has conflicts now. Could you rebase this?

Since Bigtop supports both .rpm and .deb, you need to fix files under bigtop-packages/src/deb/hadoop too. If you don't have environment to test, I can cover them in follow-up JIRA.

**Comment by qizhu-lucas** [comment]
@iwasakims I have rebased it about rpm, but the deb not updated, if you could help with deb.
Thanks.

File: bigtop-packages/src/common/hadoop/install_hadoop.sh [hunk 1]
```diff
@@ -221,6 +221,7 @@ chmod 644 ${HADOOP_DIR}/lib/*.jar ${MAPREDUCE_DIR}/lib/*.jar ${HDFS_DIR}/lib/*.j
 
 # Install webapps
 cp -ra ${BUILD_DIR}/share/hadoop/hdfs/webapps ${HDFS_DIR}/
+cp -ra ${BUILD_DIR}/share/hadoop/yarn/webapps ${YARN_DIR}/
```
Comment (iwasakims) [inline bigtop-packages/src/common/hadoop/install_hadoop.sh:224]: `./gradlew hadoop-pkg` failed with the following error.
```
+ cp -ra /home/centos/srcs/bigtop/build/hadoop/rpm/BUILD/hadoop-3.2.2-src/build/share/hadoop/yarn/webapps /home/centos/srcs/bigtop/build/hadoop/rpm/BUILDROOT/hadoop-3.2.2-1.el7.x86_64/usr/lib/hadoop-yarn/
cp: cannot stat '/home/centos/srcs/bigtop/build/hadoop/rpm/BUILD/hadoop-3.2.2-src/build/share/hadoop/yarn/webapps': No such file or directory
error: Bad exit status from /var/tmp/rpm-tmp.PviJ3q (%install)
```

Comment (qizhu-lucas) [reply bigtop-packages/src/common/hadoop/install_hadoop.sh:214]: Thanks @iwasakims for review.
I have fixed it to add -Pyarn-ui in bigtop-packages/src/common/hadoop/do-component-build.

Comment (qizhu-lucas) [reply bigtop-packages/src/common/hadoop/install_hadoop.sh:214]: Thanks @iwasakims for review.
I have fixed it to add -Pyarn-ui in bigtop-packages/src/common/hadoop/do-component-build.

File: bigtop-packages/src/common/hadoop/hadoop-hdfs-dfsrouter.svc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+HADOOP_NAME="dfsrouter"
+DAEMON="hadoop-hdfs-$HADOOP_NAME"
+DESC="Hadoop $HADOOP_NAME"
+EXEC_PATH="/usr/bin/hdfs"
```
Comment (iwasakims) [inline bigtop-packages/src/common/hadoop/hadoop-hdfs-dfsrouter.svc:18]: "/usr/lib/hadoop/sbin/hadoop-daemon.sh" must be used.

File: bigtop-packages/src/common/hadoop/hadoop-yarn-router.svc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+HADOOP_NAME="router"
+DAEMON="hadoop-yarn-$HADOOP_NAME"
+DESC="Hadoop $HADOOP_NAME"
+EXEC_PATH="/usr/bin/yarn"
```
Comment (iwasakims) [inline bigtop-packages/src/common/hadoop/hadoop-yarn-router.svc:18]: "/usr/lib/hadoop-yarn/sbin/yarn-daemon.sh" must be used.

**Comment by iwasakims** [comment]
Thanks for the update, @qizhu-lucas. While packaging works now, init scripts could not launch daemons. We should use hadoop-daemon.sh and yarn-daemon.sh to start daemons. See other `hadoop-hdfs-*.svc` and `hadoop-yarn-*.svc` as reference.

```
# systemctl start hadoop-hdfs-dfsrouter
# journalctl -xe
...
-- Unit hadoop-hdfs-dfsrouter.service has begun starting up.
Mar 01 10:55:08 148cc1d32442 runuser[31068]: pam_unix(runuser:session): session opened for user hdfs by (uid=0)
Mar 01 10:55:09 148cc1d32442 hadoop-hdfs-dfsrouter[31051]: ERROR: start is not COMMAND nor fully qualified CLASSNAME.
Mar 01 10:55:09 148cc1d32442 hadoop-hdfs-dfsrouter[31051]: Usage: hdfs [OPTIONS] SUBCOMMAND [SUBCOMMAND OPTIONS]
Mar 01 10:55:09 148cc1d32442 hadoop-hdfs-dfsrouter[31051]:   OPTIONS is none or any of:
```

File: bigtop-packages/src/common/hadoop/hadoop-yarn-router.svc (hunk 3 above)
Comment (qizhu-lucas) [reply bigtop-packages/src/common/hadoop/hadoop-yarn-router.svc]: Thanks a lot @iwasakims for patient review.
Updated it in latest pull request.

**Review by iwasakims — APPROVED** [review]
+1. The init scripts worked on CentOS 8. Thanks, @qizhu-lucas.

**Comment by iwasakims** [comment]
I filed [BIGTOP-3513](https://issues.apache.org/jira/browse/BIGTOP-3513) as a follow-up.

**Pull request merged**